client/lxd_storage_volumes.go

```diff
@@ -123,6 +123,65 @@ func (r *ProtocolLXD) GetStoragePoolVolumesAllProjects(pool string) ([]api.Stora
 	return volumes, nil
 }
 
+// GetAllVolumesWithFilter returns a filtered list of StorageVolume entries for all storage pools.
+func (r *ProtocolLXD) GetAllVolumesWithFilter(filters []string) ([]api.StorageVolume, error) {
+	err := r.CheckExtension("storage")
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.CheckExtension("list_all_storage_volumes")
+	if err != nil {
+		return nil, err
+	}
+
+	volumes := []api.StorageVolume{}
+
+	v := url.Values{}
+	v.Set("recursion", "1")
+	v.Set("filter", parseFilters(filters))
+	// Fetch the raw value
+	_, err = r.queryStruct("GET", fmt.Sprintf("/storage-volumes?%s", v.Encode()), nil, "", &volumes)
```

Comment:
Can you use `http.MethodGet` instead of `"GET"` and use URL builder instead of `fmt.Sprintf`?

Reply:
could I also change the previously existing functions that followed this pattern?